### PR TITLE
Backend/frontend infrastructure for starting a streaming job

### PR DESCRIPTION
**Pull request by swelborn**
This mostly sets up the backend with the ability to accept a job that does not have a scan ID ( a streaming job ).  It also adds this to the frontend.

`process_submit_streaming_job_event` contains a bit of duplicated code from `process_submit_job_event`. I think we can make another function that shares logic, but I'm not sure what you all think is best.

I have done work on the UI, but I did not want to include this here as it does not use redux for state management and is... messy.

### CODE REVIEW

File: backend/faust/templates/streaming.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{% include "header.j2"  %}
+{%- if machine.bbcp_dest_dir == "${DW_JOB_STRIPED}" %}
+{% include "bb_allocation_header.j2"  %}
+{%- endif %}
+
+# Remove any // ($DW_JOB_STRIPED has a trailing slash)
+BBCP_DEST_DIR=`echo "{{machine.bbcp_dest_dir}}/{{job.id}}" |  sed 's/\/\//\//g'`
+
+srun --exclusive -c {{machine.cpus_per_task}} -n {{machine.ntasks}} podman-hpc run --mpi --env HDF5_USE_FILE_LOCKING=FALSE --rm -v ${BBCP_DEST_DIR}:/output -it samwelborn:stempy-streaming /4dstem/build/zmq/node -t {{threshold}} -d {{dest_dir}} || error_exit "Error: podman execution failed."
```
Comment (cjh1) [inline backend/faust/templates/streaming.sh.j2:9]: We are currently used `shifter` for the other jobs, so unless we want to convert them over the `podman-hpc` as well, let stick with `shifter`. We should also push the image the openchemistry org for consistency.

File: backend/faust/templates/streaming.sh.j2 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% include "header.j2"  %}
+{%- if machine.bbcp_dest_dir == "${DW_JOB_STRIPED}" %}
+{% include "bb_allocation_header.j2"  %}
+{%- endif %}
+
+# Remove any // ($DW_JOB_STRIPED has a trailing slash)
+BBCP_DEST_DIR=`echo "{{machine.bbcp_dest_dir}}/{{job.id}}" |  sed 's/\/\//\//g'`
```
Comment (cjh1) [inline backend/faust/templates/streaming.sh.j2:7]: I don't think we need BBCP?

**Review by cjh1 — COMMENTED** [review]
@swelborm Are the changes innocuous if we merge them, as in will not effect the current processing path?

File: backend/faust/job_worker.py [hunk 3]
```diff
@@ -285,11 +289,62 @@ async def update_slurm_job_id(
...
+    await update_slurm_job_id(session, event.job.id, slurm_id)
+    
 
 async def process_submit_job_event(
     session: aiohttp.ClientSession, event: SubmitJobEvent
 ) -> None:
 
+    if event.scan is None:
+        await process_submit_streaming_job_event(session, event)
+        return None
```
[the remark this answers is not in the document]
Comment (swelborn) [reply backend/faust/job_worker.py:346]: The rest of `job_worker.py` has `return None` statements, I think I was trying to stick to the convention (I figured it had something to do with static typing). I can change it, just let me know.

[the remark this answers is not in the document]
Comment (cjh1) [reply backend/faust/job_worker.py:346]: Fair enough, normally I would only use `return None` if the function does actually return something ( as in its signature has a return type other than `None` ).

File: backend/faust/templates/streaming.sh.j2 (hunk 2 above)
Comment (swelborn) [reply backend/faust/templates/streaming.sh.j2]: Added `zmq_dest_dir` to Machine.

**Comment by swelborn** [comment]
Tested and does not appear to impact normal operations.

**Comment by swelborn** [comment]
Superseded by #823

**Pull request closed without merging**